src/currencymanager.cpp

```diff
@@ -88,47 +104,85 @@ int CurrencyManager::TotalWisdomValue() {
 
 void CurrencyManager::ClearCurrency() {
     for (auto& currency : currencies_) {
-        currency.count = 0;
+        currency->count = 0;
     }
     for (auto& wisdom : wisdoms_) {
         wisdom = 0;
     }
 }
-
 void CurrencyManager::InitCurrency() {
+    Deserialize(data_.Get("currency_items"), &currencies_);
+    for (unsigned int i = 0; i < CurrencyWisdomValue.size(); i++) {
+        wisdoms_.push_back(0);
+    }
+}
+
+void CurrencyManager::FirstInitCurrency() {
     std::string value = "";
-    for (auto curr : CurrencyAsString) {
+    //Dummy items + dummy currency_last_value
+    //TODO : can i get the size of the Currency enum ??
+    for(unsigned int i = 0; i < CurrencyAsString.size();i++) {
+        currencies_.push_back(std::make_shared<CurrencyItem>(0, static_cast<Currency>(i), 1, 1));
+
         value += "0;";
     }
     value.pop_back(); // Remove the last ";"
-    data_.Set("currency_base", value);
+    data_.Set("currency_items", Serialize(currencies_));
     data_.Set("currency_last_value", value);
+    data_.SetBool("currency_show_chaos", true);
+    data_.SetBool("currency_show_exalt", true);
 }
 
-void CurrencyManager::LoadCurrency() {
+void CurrencyManager::MigrateCurrency() {
     std::vector<std::string> list = Util::StringSplit(data_.Get("currency_base"), ';');
-    for (unsigned int i = 0; i < list.size(); i++) {
-        CurrencyItem curr;
-        curr.count = 0;
-        curr.name = CurrencyAsString[i];
-        // We can't use the toDouble function from QT, because it encodes a double with a ","
-        // Might be related to localisation issue, but anyway it's safer this way
-        curr.base = std::stod(list[i]);
-        curr.exalt = 0;
-        currencies_.push_back(curr);
-    }
-    for (unsigned int i = 0; i < CurrencyWisdomValue.size(); i++) {
-        wisdoms_.push_back(0);
+        for (unsigned int i = 0; i < list.size(); i++) {
+            // We can't use the toDouble function from QT, because it encodes a double with a ","
```

Comment:
Since you're french, you'll get the "," (assuming your computer localization is French/Belgian/Canadian/some other francophonie locale), but you can get around this by using: http://doc.qt.io/qt-5/qlocale.html#toDouble